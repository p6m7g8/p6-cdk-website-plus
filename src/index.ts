export * from './p6-cdk-website-plus'
