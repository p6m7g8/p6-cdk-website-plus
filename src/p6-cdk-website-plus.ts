import type { Construct } from 'constructs'
import * as cdk from 'aws-cdk-lib'
import * as certificatemanager from 'aws-cdk-lib/aws-certificatemanager'
import * as cloudfront from 'aws-cdk-lib/aws-cloudfront'
import * as cloudfront_origins from 'aws-cdk-lib/aws-cloudfront-origins'
import * as iam from 'aws-cdk-lib/aws-iam'
import * as route53 from 'aws-cdk-lib/aws-route53'
import * as route53Patterns from 'aws-cdk-lib/aws-route53-patterns'
import * as route53targets from 'aws-cdk-lib/aws-route53-targets'
import * as s3 from 'aws-cdk-lib/aws-s3'

export interface IP6CDKWebsiteProps {
  hostedZoneName: string
  verifyEmail: string
  cloudfrontRecordName: string
}

export class P6CDKWebsitePlus extends cdk.Resource {
  constructor(scope: Construct, id: string, props: IP6CDKWebsiteProps) {
    super(scope, id)

    // Lookup for the hosted zone
    const hostedZone = route53.HostedZone.fromLookup(this, 'HostedZone', {
      domainName: props.hostedZoneName,
    })

    // Create the certificate for the domain
    const certificate = new certificatemanager.Certificate(this, 'Certificate', {
      domainName: props.hostedZoneName,
      subjectAlternativeNames: [props.cloudfrontRecordName],
      validation: certificatemanager.CertificateValidation.fromEmail({
        email: props.verifyEmail,
      }),
    })

    // Define the S3 bucket for website hosting
    const bucket = new s3.Bucket(this, 'MyBucket', {
      accessControl: s3.BucketAccessControl.BUCKET_OWNER_FULL_CONTROL,
      publicReadAccess: true, // Required for static website hosting to work with CloudFront
      websiteIndexDocument: 'index.html',
      removalPolicy: cdk.RemovalPolicy.DESTROY,
      cors: [
        {
          allowedMethods: [s3.HttpMethods.GET, s3.HttpMethods.PUT],
          allowedOrigins: ['*'],
          allowedHeaders: ['*'],
          maxAge: 3000,
        },
      ],
      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ACLS, // Allows public access via bucket policy
    })

    const logBucket = new s3.Bucket(this, 'LogBucket', {
      removalPolicy: cdk.RemovalPolicy.DESTROY,
      encryption: s3.BucketEncryption.S3_MANAGED,
      publicReadAccess: false,
      accessControl: s3.BucketAccessControl.LOG_DELIVERY_WRITE, // Enable ACLs for CloudFront logging
      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ACLS, // Allow ACLs for logging while blocking public access
    })

    // Grant CloudFront permission to write logs to the log bucket
    logBucket.addToResourcePolicy(new iam.PolicyStatement({
      effect: iam.Effect.ALLOW,
      actions: ['s3:PutObject'],
      resources: [logBucket.arnForObjects('*')],
      principals: [new iam.ServicePrincipal('delivery.logs.amazonaws.com')],
      conditions: {
        StringEquals: {
          'aws:SourceAccount': cdk.Aws.ACCOUNT_ID,
        },
        ArnLike: {
          'aws:SourceArn': `arn:aws:cloudfront::${cdk.Aws.ACCOUNT_ID}:distribution/*`,
        },
      },
    }))

    const cachePolicy = new cloudfront.CachePolicy(this, 'CachePolicy', {
      defaultTtl: cdk.Duration.days(1),
      minTtl: cdk.Duration.seconds(0),
      maxTtl: cdk.Duration.days(365),
      cookieBehavior: cloudfront.CacheCookieBehavior.none(),
      headerBehavior: cloudfront.CacheHeaderBehavior.none(),
      queryStringBehavior: cloudfront.CacheQueryStringBehavior.none(),
      enableAcceptEncodingGzip: true,
      enableAcceptEncodingBrotli: true,
    })

    const distribution = new cloudfront.Distribution(this, 'Distribution', {
      defaultRootObject: 'index.html',
      comment: props.cloudfrontRecordName,
      enabled: true,
      priceClass: cloudfront.PriceClass.PRICE_CLASS_100,
      domainNames: [props.cloudfrontRecordName], // Use only one domain to avoid conflicts
      certificate,
      minimumProtocolVersion: cloudfront.SecurityPolicyProtocol.TLS_V1_2_2021,
      defaultBehavior: {
        origin: new cloudfront_origins.S3StaticWebsiteOrigin(bucket),
        allowedMethods: cloudfront.AllowedMethods.ALLOW_GET_HEAD,
        viewerProtocolPolicy: cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
        cachePolicy,
      },
      enableLogging: true,
      logBucket,
      logIncludesCookies: true,
    })

    const cloudfrontTarget = route53.RecordTarget.fromAlias(
      new route53targets.CloudFrontTarget(distribution),
    )

    // Create both A (IPv4) and AAAA (IPv6) DNS records for www.gollucci.com (cloudfrontRecordName)
    new route53.ARecord(this, 'CloudfrontDnsRecordWWW', {
      zone: hostedZone,
      recordName: props.cloudfrontRecordName,
      target: cloudfrontTarget,
    })

    new route53.AaaaRecord(this, 'CloudfrontDnsRecordAAAAWWW', {
      zone: hostedZone,
      recordName: props.cloudfrontRecordName,
      target: cloudfrontTarget,
    })

    // Use HttpsRedirect for redirecting root domain (hostedZoneName) to www
    new route53Patterns.HttpsRedirect(this, 'Redirect', {
      recordNames: [props.hostedZoneName], // Redirect root domain
      targetDomain: props.cloudfrontRecordName, // Redirect to www
      zone: hostedZone,
    })
  }
}
