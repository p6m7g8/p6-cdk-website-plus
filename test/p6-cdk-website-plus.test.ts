import * as cdk from 'aws-cdk-lib'
import { Template } from 'aws-cdk-lib/assertions'
import { P6CDKWebsitePlus } from '../src'

it('p6CDKWebsitePlus snapshot test', () => {
  // GIVEN
  const theEnv = {
    account: process.env.CDK_DEPLOY_ACCOUNT || process.env.CDK_DEFAULT_ACCOUNT,
    region: process.env.CDK_DEPLOY_REGION || process.env.CDK_DEFAULT_REGION,
  }
  const app = new cdk.App()
  const stack = new cdk.Stack(app, 'MyStack', { env: theEnv })

  // WHEN
  new P6CDKWebsitePlus(stack, 'p6-cdk-website-plus', {
    hostedZoneName: 'p6m7g8.org',
    verifyEmail: 'test@p6m7g8.org',
    cloudfrontRecordName: 'www.p6m7g8.org',
  })

  // THEN
  const template = Template.fromStack(stack).toJSON()
  expect(template).toMatchSnapshot()
})
