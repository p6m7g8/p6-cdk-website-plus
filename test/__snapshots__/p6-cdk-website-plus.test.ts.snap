// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`p6CDKWebsitePlus snapshot test 1`] = `
{
  "Mappings": {
    "AWSCloudFrontPartitionHostedZoneIdMap": {
      "aws": {
        "zoneId": "Z2FDTNDATAQYW2",
      },
      "aws-cn": {
        "zoneId": "Z3RFFRIM2A3IF5",
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "p6cdkwebsiteplusCachePolicyB3E25F40": {
      "Properties": {
        "CachePolicyConfig": {
          "DefaultTTL": 86400,
          "MaxTTL": 31536000,
          "MinTTL": 0,
          "Name": "MyStackp6cdkwebsiteplusCachePolicyEDFF9D09-us-east-1",
          "ParametersInCacheKeyAndForwardedToOrigin": {
            "CookiesConfig": {
              "CookieBehavior": "none",
            },
            "EnableAcceptEncodingBrotli": true,
            "EnableAcceptEncodingGzip": true,
            "HeadersConfig": {
              "HeaderBehavior": "none",
            },
            "QueryStringsConfig": {
              "QueryStringBehavior": "none",
            },
          },
        },
      },
      "Type": "AWS::CloudFront::CachePolicy",
    },
    "p6cdkwebsiteplusCertificate363867BF": {
      "Properties": {
        "DomainName": "p6m7g8.org",
        "DomainValidationOptions": [
          {
            "DomainName": "p6m7g8.org",
            "ValidationDomain": "p6m7g8.org",
          },
          {
            "DomainName": "www.p6m7g8.org",
            "ValidationDomain": "p6m7g8.org",
          },
        ],
        "SubjectAlternativeNames": [
          "www.p6m7g8.org",
        ],
        "Tags": [
          {
            "Key": "Name",
            "Value": "MyStack/p6-cdk-website-plus/Certificate",
          },
        ],
        "ValidationMethod": "EMAIL",
      },
      "Type": "AWS::CertificateManager::Certificate",
    },
    "p6cdkwebsiteplusCloudfrontDnsRecordAAAAWWWD3670959": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "p6cdkwebsiteplusDistribution75B33D1D",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "www.p6m7g8.org.",
        "Type": "AAAA",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "p6cdkwebsiteplusCloudfrontDnsRecordWWW266C1732": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "p6cdkwebsiteplusDistribution75B33D1D",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "www.p6m7g8.org.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "p6cdkwebsiteplusDistribution75B33D1D": {
      "Properties": {
        "DistributionConfig": {
          "Aliases": [
            "p6m7g8.org",
            "www.p6m7g8.org",
          ],
          "Comment": "www.p6m7g8.org",
          "DefaultCacheBehavior": {
            "AllowedMethods": [
              "GET",
              "HEAD",
            ],
            "CachePolicyId": {
              "Ref": "p6cdkwebsiteplusCachePolicyB3E25F40",
            },
            "Compress": true,
            "TargetOriginId": "MyStackp6cdkwebsiteplusDistributionOrigin11194C0BD",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Logging": {
            "Bucket": {
              "Fn::GetAtt": [
                "p6cdkwebsiteplusLogBucket1DBBB6EA",
                "RegionalDomainName",
              ],
            },
            "IncludeCookies": true,
          },
          "Origins": [
            {
              "CustomOriginConfig": {
                "OriginProtocolPolicy": "http-only",
                "OriginSSLProtocols": [
                  "TLSv1.2",
                ],
              },
              "DomainName": {
                "Fn::Select": [
                  2,
                  {
                    "Fn::Split": [
                      "/",
                      {
                        "Fn::GetAtt": [
                          "p6cdkwebsiteplusMyBucket91F315D6",
                          "WebsiteURL",
                        ],
                      },
                    ],
                  },
                ],
              },
              "Id": "MyStackp6cdkwebsiteplusDistributionOrigin11194C0BD",
            },
          ],
          "PriceClass": "PriceClass_100",
          "ViewerCertificate": {
            "AcmCertificateArn": {
              "Ref": "p6cdkwebsiteplusCertificate363867BF",
            },
            "MinimumProtocolVersion": "TLSv1.2_2021",
            "SslSupportMethod": "sni-only",
          },
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "p6cdkwebsiteplusLogBucket1DBBB6EA": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "AccessControl": "LogDeliveryWrite",
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "OwnershipControls": {
          "Rules": [
            {
              "ObjectOwnership": "ObjectWriter",
            },
          ],
        },
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "IgnorePublicAcls": true,
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "p6cdkwebsiteplusLogBucketPolicy866A9602": {
      "Properties": {
        "Bucket": {
          "Ref": "p6cdkwebsiteplusLogBucket1DBBB6EA",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:PutObject",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudfront::",
                        {
                          "Ref": "AWS::AccountId",
                        },
                        ":distribution/*",
                      ],
                    ],
                  },
                },
                "StringEquals": {
                  "aws:SourceAccount": {
                    "Ref": "AWS::AccountId",
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "Service": "delivery.logs.amazonaws.com",
              },
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    {
                      "Fn::GetAtt": [
                        "p6cdkwebsiteplusLogBucket1DBBB6EA",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "p6cdkwebsiteplusMyBucket91F315D6": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "AccessControl": "BucketOwnerFullControl",
        "CorsConfiguration": {
          "CorsRules": [
            {
              "AllowedHeaders": [
                "*",
              ],
              "AllowedMethods": [
                "GET",
                "PUT",
              ],
              "AllowedOrigins": [
                "*",
              ],
              "MaxAge": 3000,
            },
          ],
        },
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "IgnorePublicAcls": true,
        },
        "WebsiteConfiguration": {
          "IndexDocument": "index.html",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "p6cdkwebsiteplusMyBucketPolicyE7D8D7E9": {
      "Properties": {
        "Bucket": {
          "Ref": "p6cdkwebsiteplusMyBucket91F315D6",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Principal": {
                "CanonicalUser": {
                  "Fn::GetAtt": [
                    "p6cdkwebsiteplusOAIC7AC1AD0",
                    "S3CanonicalUserId",
                  ],
                },
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "p6cdkwebsiteplusMyBucket91F315D6",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "p6cdkwebsiteplusMyBucket91F315D6",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "p6cdkwebsiteplusOAIC7AC1AD0": {
      "Properties": {
        "CloudFrontOriginAccessIdentityConfig": {
          "Comment": "Allows CloudFront to reach the bucket",
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
    "p6cdkwebsiteplusRedirectRedirectAliasRecord0c38695AA6882E": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "p6cdkwebsiteplusRedirectRedirectDistributionCFDistribution541F087D",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "p6m7g8.org.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "p6cdkwebsiteplusRedirectRedirectAliasRecordSix0c38693DBF8242": {
      "Properties": {
        "AliasTarget": {
          "DNSName": {
            "Fn::GetAtt": [
              "p6cdkwebsiteplusRedirectRedirectDistributionCFDistribution541F087D",
              "DomainName",
            ],
          },
          "HostedZoneId": {
            "Fn::FindInMap": [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "p6m7g8.org.",
        "Type": "AAAA",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "p6cdkwebsiteplusRedirectRedirectBucketAE17B5E0": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "WebsiteConfiguration": {
          "RedirectAllRequestsTo": {
            "HostName": "www.p6m7g8.org",
            "Protocol": "https",
          },
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "p6cdkwebsiteplusRedirectRedirectCertificateCertificateRequestorFunctionC85F0BF4": {
      "DependsOn": [
        "p6cdkwebsiteplusRedirectRedirectCertificateCertificateRequestorFunctionServiceRoleDefaultPolicyD4594615",
        "p6cdkwebsiteplusRedirectRedirectCertificateCertificateRequestorFunctionServiceRole78E26D4B",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": "cdk-hnb659fds-assets-346733622331-us-east-1",
          "S3Key": "b073cebcf4d61fb152a30f5a5e57a94df7f980a549fdf1a79a0b18c5750522d8.zip",
        },
        "Handler": "index.certificateRequestHandler",
        "Role": {
          "Fn::GetAtt": [
            "p6cdkwebsiteplusRedirectRedirectCertificateCertificateRequestorFunctionServiceRole78E26D4B",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "p6cdkwebsiteplusRedirectRedirectCertificateCertificateRequestorFunctionServiceRole78E26D4B": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "p6cdkwebsiteplusRedirectRedirectCertificateCertificateRequestorFunctionServiceRoleDefaultPolicyD4594615": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "acm:RequestCertificate",
                "acm:DescribeCertificate",
                "acm:DeleteCertificate",
                "acm:AddTagsToCertificate",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": "route53:GetChange",
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": "route53:changeResourceRecordSets",
              "Condition": {
                "ForAllValues:StringEquals": {
                  "route53:ChangeResourceRecordSetsActions": [
                    "UPSERT",
                  ],
                  "route53:ChangeResourceRecordSetsRecordTypes": [
                    "CNAME",
                  ],
                },
                "ForAllValues:StringLike": {
                  "route53:ChangeResourceRecordSetsNormalizedRecordNames": [
                    "*.p6m7g8.org",
                    "*.p6m7g8.org",
                  ],
                },
              },
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":route53:::hostedzone/DUMMY",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "p6cdkwebsiteplusRedirectRedirectCertificateCertificateRequestorFunctionServiceRoleDefaultPolicyD4594615",
        "Roles": [
          {
            "Ref": "p6cdkwebsiteplusRedirectRedirectCertificateCertificateRequestorFunctionServiceRole78E26D4B",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "p6cdkwebsiteplusRedirectRedirectCertificateCertificateRequestorResource489D7530": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "DomainName": "p6m7g8.org",
        "HostedZoneId": "DUMMY",
        "Region": "us-east-1",
        "ServiceToken": {
          "Fn::GetAtt": [
            "p6cdkwebsiteplusRedirectRedirectCertificateCertificateRequestorFunctionC85F0BF4",
            "Arn",
          ],
        },
        "SubjectAlternativeNames": [
          "p6m7g8.org",
        ],
      },
      "Type": "AWS::CloudFormation::CustomResource",
      "UpdateReplacePolicy": "Delete",
    },
    "p6cdkwebsiteplusRedirectRedirectDistributionCFDistribution541F087D": {
      "Properties": {
        "DistributionConfig": {
          "Aliases": [
            "p6m7g8.org",
          ],
          "Comment": "Redirect to www.p6m7g8.org from p6m7g8.org",
          "DefaultCacheBehavior": {
            "AllowedMethods": [
              "GET",
              "HEAD",
            ],
            "CachedMethods": [
              "GET",
              "HEAD",
            ],
            "Compress": true,
            "ForwardedValues": {
              "Cookies": {
                "Forward": "none",
              },
              "QueryString": false,
            },
            "TargetOriginId": "origin1",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": [
            {
              "ConnectionAttempts": 3,
              "ConnectionTimeout": 10,
              "CustomOriginConfig": {
                "HTTPPort": 80,
                "HTTPSPort": 443,
                "OriginKeepaliveTimeout": 5,
                "OriginProtocolPolicy": "http-only",
                "OriginReadTimeout": 30,
                "OriginSSLProtocols": [
                  "TLSv1.2",
                ],
              },
              "DomainName": {
                "Fn::Select": [
                  2,
                  {
                    "Fn::Split": [
                      "/",
                      {
                        "Fn::GetAtt": [
                          "p6cdkwebsiteplusRedirectRedirectBucketAE17B5E0",
                          "WebsiteURL",
                        ],
                      },
                    ],
                  },
                ],
              },
              "Id": "origin1",
            },
          ],
          "PriceClass": "PriceClass_All",
          "ViewerCertificate": {
            "AcmCertificateArn": {
              "Fn::GetAtt": [
                "p6cdkwebsiteplusRedirectRedirectCertificateCertificateRequestorResource489D7530",
                "Arn",
              ],
            },
            "SslSupportMethod": "sni-only",
          },
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
